packages/apollo-fetch-upload/src/index.ts

```diff
@@ -0,0 +1,37 @@
+import {
+  createApolloFetch,
+  constructDefaultOptions,
+  // Types:
+  ApolloFetch,
+  GraphQLRequest,
+} from 'apollo-fetch';
+import { extractFiles } from 'extract-files';
+
+export function constructUploadOptions(
+  requestOrRequests: GraphQLRequest | GraphQLRequest[],
+  options: RequestInit,
+): RequestInit {
+  const files = extractFiles(requestOrRequests);
```

Comment:
@evanshauser Does it matter that `requestOrRequests` is modified by `extractFiles`? Or would it be better if extract files did not modify the input:

```
const { files, modifiedTree } = extractFiles(requestOrRequests)
```

Keep in mind that there are performance considerations to messing around with objects that contain potentially megabytes of files.